Source/WebCore/editing/TypingCommand.cpp

```diff
@@ -458,10 +458,10 @@ void TypingCommand::markMisspellingsAfterTyping(Type commandType)
         VisiblePosition p2 = startOfWord(start, LeftWordIfOnBoundary);
         if (p1 != p2) {
             auto range = makeSimpleRange(p1, p2);
-            String strippedPreviousWord;
+            StringView trimmedPreviousWord;
             if (range && (commandType == TypingCommand::Type::InsertText || commandType == TypingCommand::Type::InsertLineBreak || commandType == TypingCommand::Type::InsertParagraphSeparator || commandType == TypingCommand::Type::InsertParagraphSeparatorInQuotedContent))
-                strippedPreviousWord = plainText(*range).stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline);
-            document().editor().markMisspellingsAfterTypingToWord(p1, endingSelection(), !strippedPreviousWord.isEmpty());
+                trimmedPreviousWord = StringView(plainText(*range)).trim(deprecatedIsSpaceOrNewline);
```

Comment:
This isn't safe

Reply:
Interesting and sorry about that. Will revert that part.

Reply:
To clarify, `plainText(*range)` returns a new temporary String. Then you make a StringView to this temporary, then you call trim() which will trim your StringView and returns a StringView as well. On the next line, the string returned by plainText() is dead and your `trimmedPreviousWord` StringView points to a dead String.